### ISSUE TITLE
How can I avoid escapes in json?

### ISSUE THREAD

**Issue by Nick-Gatti** [huginn/huginn#3413]
Trying to make a POST call to a JSON-RPC API and it doesn't seem to like the escaped quotation marks in my payload. is there a way around this? Example call:

```
{
  "jsonrpc": "2.0",
  "method": "getUserSubscriptions",
  "params": [
    "123456789"
  ],
  "id": 1
}
```

which gets output as

```
{
  \"jsonrpc\": \"2.0\",
  \"method\": \"getUserSubscriptions\",
  \"params\": [
    \"123456789\"
  ],
  \"id\": 1
}
```

**Comment by virtadpt** [comment]
Can you post the entire agent (suitably redacted)?

**Comment by Nick-Gatti** [comment]
`{
  "endpoint": "https://api.<url>.net",
  "expected_receive_period_in_days": "7",
  "content_type": "json",
  "method": "post",
  "payload": {
    "jsonrpc": "2.0",
    "method": "getUserSubscriptions",
    "params": [
      "123456789"
    ],
    "id": 1
  },
  "headers": {
  },
  "emit_events": "true",
  "no_merge": "false",
  "output_mode": "clean",
  "log_requests": "true"
}`

**Comment by Nick-Gatti** [comment]
```
{
  "post_url": "https://api.<url>.net/",
  "expected_update_period_in_days": "1",
  "content_type": "json",
  "method": "post",
  "headers": {
    "Content-Type": "application/json"
  },
  "payload": {
    "jsonrpc": "2.0",
    "method": "getUserSubscriptions",
    "params": [
      "123456789"
    ],
    "id": 1
  },
  "emit_events": true,
  "no_merge": true,
  "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:125.0) Gecko/20100101 Firefox/125.0"
}
```

**Comment by Unending** [comment]
I can't reproduce this.

Are you sure that's what actually being send or is that just what huginn shows as the outgoing event since you enabled `emit_events`?

**Comment by Nick-Gatti** [comment]
I can't seem to find any way of actually telling what Huginn is sending- I'm admittedly assuming that that's what's happening since running the same payload through Postman is successful. The response is pretty ugly as well, so I'm not sure exactly what's going on on the server's side either-

```
{
  "body": "ERROR 500",
  "status": 200,
  "headers": {
...
  }
}
```

**Comment by Nick-Gatti** [comment]
Actually, I did test it by sending it to some mock endpoint, now that I think about it- it's been a couple weeks since I messed around with this.

**Comment by virtadpt** [comment]
Okay, HTTP 500 is an internal server error (https://http.cat/500).  This might not be a problem on your end.

Looking at how you have your payload set up in context, it looks right.  Consistent with my own agents, anyway.

What happens if you use cURL to send the same kind of request with the -vv option?

**Comment by Nick-Gatti** [comment]
Sending the same thing works fine:

```
curl --request POST \
  --url https://api.<url>.net/ \
  --header 'content-type: application/json' \
  --data '{
  "jsonrpc": "2.0",
  "method": "getUserSubscriptions",
  "params": [
    "123456789"
  ],
  "id": 1
}'
```

I used [ptsv3](http://ptsv3.com/) to test what exactly was getting sent (This was from a few weeks back):

curl/Postman:
`{'id': 12828, 'date': '2024-05-13', 'time': '18:53:23', 'method': 'POST', 'body': '{"jsonrpc":"2.0","method":"getUserSubscriptions","params":[""],"id":1}'}
`

Huginn:
` {'id': 99423, 'date': '2024-05-13', 'time': '18:55:48', 'method': 'POST', 'body': '{\n "jsonrpc": "2.0",\n "method": "getUserSubscriptions",\n "params": [\n ""\n ],\n "id": 1\n}'}]`

**Comment by Nick-Gatti** [comment]
Weird, just tried again and it's making a liar out of me.

**Comment by knu** [comment]
Did you disable the `parse_json` option?

**Comment by Nick-Gatti** [comment]
Yeah, I did. No difference.

**Comment by knu** [comment]
You shouldn't.  Just as instructed as follows, without `parse_body` enabled the payload is a raw text.

> If the parse_body option is set to true and the content type of the response is JSON, it is parsed to a JSON object. Otherwise it is raw text.

**Comment by Nick-Gatti** [comment]
I'm getting the response back fine, something's going on with the request- just not sure what.

**Comment by Nick-Gatti** [comment]
Wound up just using a cURL shell command. Not ideal, but it works.

**Comment by knu** [comment]
I created a test endpoint that accepts a POST request and returns a JSON body, and a PostAgent with the following options successfully created an event with a JSON object body:

```json
{
  "post_url": "https://post-test-huginn.akinori.workers.dev/",
  "method": "post",
  "emit_events": "true",
  "parse_body": "true"
}
```

Emitted event:
```json
{
  "body": {
    "message": "hello, world.",
    "method": "POST"
  },
  "status": 200,
  "headers": {
    "Date": "Fri, 07 Jun 2024 06:37:08 GMT",
    "Content-Type": "application/json",
    "Content-Length": "43",
    "Report-To": "{\"endpoints\":[{\"url\":\"https:\\/\\/a.nel.cloudflare.com\\/report\\/v4?s=%2BvygRLUgUfGDmpzAM8xaD4Q2tQHQANy8m0SSJub7EtFOJzC4naJv6nNrp582FVTCSrZKgsi%2F8oKPUJdMh73HhNpBMl5YVnvIEs5bcO7oEwzSoT2ftLwuWw%2FvQCDHlsFxLz4LUcfjGHIudZ2EdXpAhxIMD6IwATg%3D\"}],\"group\":\"cf-nel\",\"max_age\":604800}",
    "Nel": "{\"success_fraction\":0,\"report_to\":\"cf-nel\",\"max_age\":604800}",
    "Vary": "Accept-Encoding",
    "Server": "cloudflare",
    "Cf-Ray": "88fea1c0dbfd7341-NRT",
    "Alt-Svc": "h3=\":443\"; ma=86400"
  }
}
```

The `parse_body` option has been available since July 2023.

**Comment by Nick-Gatti** [comment]
Yeah, the response JSON worked fine for me. It was the request JSON that was getting malformed/formatting weird somehow.
